tests/phpunit/tests/functions/wpCheckFiletype.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * Test the wp_upload_bits function
+ *
+ * @group Functions
+ * @group Upload
+ * @covers ::wp_check_filetype
+ */
+class Test_wp_check_filetype extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 57151
+	 * @dataProvider wp_check_filetype_dataset
+	 */
+	function test_wp_check_filetypes( $filename, $mines, $expected ) {
+		$this->assertSame( $expected, wp_check_filetype( $filename, $mines ) );
+	}
+
+	function wp_check_filetype_dataset() {
+		return array(
+			'default'     => array(
+				'filename' => 'canola.jpg',
+				'mines'    => null,
```

Comment:
```suggestion
				'mimes'    => null,
```
While this dataset covers the defaults (i.e. `null`), I'd also suggest an additional dataset that sets `$mimes` to `array()` to reflect an `empty()` value of the documented datatype.